lib/discovery.dart

```diff
@@ -152,10 +152,30 @@ Future<Packages> findPackagesFromNonFile(Uri nonFileUri,
   });
 }
 
-/// Fetches a file using the http library.
+/// Fetches a file over http.
 Future<List<int>> _httpGet(Uri uri) {
-  return http.get(uri).then((http.Response response) {
-    if (response.statusCode == 200) return response.bodyBytes;
-    throw 0;  // The error message isn't being used for anything.
+  HttpClient client = new HttpClient();
+  return client.getUrl(uri)
+  .then((HttpClientRequest request) => request.close())
```

Comment:
No, I copied it from https://github.com/dart-lang/sdk/blob/master/pkg/compiler/lib/src/source_file_provider.dart#L71
I'll format this file